src/neat/neat_functions.py

```diff
@@ -92,3 +92,64 @@ def delta(genome1: Genome, genome2: Genome, c1: float = 1.0, c2: float = 1.0, c3
     delta = c1 * (excess / n) + c2 * (disjoint / n) + c3 * (total_diff / matching)
 
     return delta
+
+def mutate(g: Genome, ino: int, get_fitness: Callable):
```

Comment:
We should split this into three functions.

mutate_weights
add_node
add_connection. Each should take in a list of genes and output a list of genes rather than Genome to Genome.

Reply:
This is easy to do for the mutate_weights. But in the add node and add connection, I use self variables from the Genome like the set of all nodes, set of directed connections, and I also update these variables after each mutation. I could have all of these as inputs, but it seems like passing in the Genome is easier.

Reply:
This is easy to do for mutate_weights. But for the structural mutations, I use self variables from the Genome class like the set of all nodes, and the set of directed connections. I also update all the self variables like the set of innovation numbers after each mutation changes the genome. I could pass all of these variables into these functions, but it seems easier to pass in the Genome now that I look at it.